homeassistant/components/vodafone_station/api.py

```diff
@@ -0,0 +1,396 @@
+        self._mac = mac
+        self._name = name
+        self._wifi: str | None = None
+
+    def update(
+        self,
+        dev_info: Device,
+        consider_home: float = DEFAULT_CONSIDER_HOME.total_seconds(),
+    ) -> None:
+        """Update device info."""
+        utc_point_in_time = dt_util.utcnow()
+
+        if self._last_activity:
+            consider_home_evaluated = (
+                utc_point_in_time - self._last_activity
+            ).total_seconds() < consider_home
+        else:
+            consider_home_evaluated = dev_info.connected
+
+        if not self._name:
+            self._name = dev_info.name or self._mac.replace(":", "_")
+
+        self._connected = dev_info.connected or consider_home_evaluated
+
+        if dev_info.connected:
+            self._last_activity = utc_point_in_time
+
+        self._connection_type = dev_info.connection_type
+        self._ip_address = dev_info.ip_address
+        self._wifi = dev_info.wifi
+
+    @property
+    def connection_type(self) -> str:
+        """Return connected status."""
+        return self._connection_type or ""
+
+    @property
+    def is_connected(self) -> bool:
+        """Return connected status."""
+        return self._connected
+
+    @property
+    def mac_address(self) -> str:
+        """Get MAC address."""
+        return self._mac
+
+    @property
+    def hostname(self) -> str | None:
+        """Get Name."""
+        return self._name
+
+    @property
+    def ip_address(self) -> str | None:
+        """Get IP address."""
+        return self._ip_address
+
+    @property
+    def last_activity(self) -> datetime | None:
+        """Return device last activity."""
+        return self._last_activity
+
+    @property
+    def wifi(self) -> str | None:
+        """Return device WIFi connection."""
+        return self._wifi
+
+
+class VodafoneStationApi(DataUpdateCoordinator):
+    """Queries router running Vodafone Station firmware."""
+
+    def __init__(
+        self, host: str, ssl: bool, username: str, password: str, hass: HomeAssistant
+    ) -> None:
+        """Initialize the scanner."""
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=f"{DOMAIN}-{host}-coordinator",
+            update_interval=timedelta(seconds=30),
+        )
+
+        self.host = host
+        self.protocol = "https" if ssl else "http"
+        self.username = username
+        self.password = password
+        self.hass = hass
+        self.base_url = f"{self.protocol}://{self.host}"
+        self.headers = {
+            "User-Agent": "Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0",
+            "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
+            "Accept-Language": "en-GB,en;q=0.5",
+            "Origin": self.base_url,
+            "Referer": f"{self.base_url}/login.html",
+            "DNT": "1",
+        }
```

Comment:
All of this should be in an external lib

Reply:
moved to aiovodafone